### PR TITLE
Fix : Use explicit JOINS on lists instead of comma join

### PR DESCRIPTION
**Pull request by thomas-Ngr**
Fix error when using hook printFieldListFrom in a module.

One should not use comma-joins, as stated in the MySQL manual:
"The precedence of the comma operator is less than that of INNER JOIN, CROSS JOIN, LEFT JOIN, and so on. If you mix comma joins with the other join types when there is a join condition, an error of the form Unknown column 'col_name' in 'on clause' may occur. Information about dealing with this problem is given later in this section. " [source](https://dev.mysql.com/doc/refman/8.0/en/join.html)

In a PrintFieldListFrom hook on facture list, I add this statement : `LEFT JOIN llx_element_contact ec ON ec.element_id = f.rowid` and I get this error : `Unknown column 'f.rowid' in 'on clause'`

### CODE REVIEW

File: htdocs/comm/propal/list.php [hunk 1]
```diff
@@ -548,22 +548,22 @@
 $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_availability as ava on (ava.rowid = p.fk_availability)";
 // We'll need this table joined to the select in order to filter by sale
 if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) {
-	$sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
+	$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON s.rowid = sc.fk_soc";
```
Comment (eldy) [inline htdocs/comm/propal/list.php:551]: Left join has not the same result than exact join (inner join)

File: htdocs/commande/list.php [hunk 2]
```diff
@@ -449,7 +449,7 @@
 if (($search_categ_cus > 0) || ($search_categ_cus == -2)) {
 	$sql .= ' LEFT JOIN '.MAIN_DB_PREFIX."categorie_societe as cc ON s.rowid = cc.fk_soc"; // We'll need this table joined to the select in order to filter by categ
 }
-$sql .= ', '.MAIN_DB_PREFIX.'commande as c';
+$sql .= 'RIGHT JOIN '.MAIN_DB_PREFIX.'commande as c ON c.fk_soc = s.rowid';
```
Comment (eldy) [inline htdocs/commande/list.php:452]: We must avoid right join that generated a very difficult way to understand readability of requests. Only inner and left join should be used.

File: htdocs/commande/list.php [hunk 3]
```diff
@@ -464,19 +464,19 @@
...
+	$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_type_contact as tc ON ec.fk_c_type_contact = tc.rowid AND tc.element='commande' AND tc.source='internal'";
 }
 
 // Add table from hooks
 $parameters = array();
 $reshook = $hookmanager->executeHooks('printFieldListFrom', $parameters, $object); // Note that $action and $object may have been modified by hook
 $sql .= $hookmanager->resPrint;
 
-$sql .= ' WHERE c.fk_soc = s.rowid';
+$sql .= ' WHERE c.rowid IS NOT NULL';
```
Comment (eldy) [inline htdocs/commande/list.php:479]: This generates very bad performance and can be avoid by using inner join instead of right join.

**Comment by hregis** [comment]
@thomas-Ngr just for fun 😉

File: htdocs/comm/propal/list.php (hunk 1 above)
Comment (thomas-Ngr) [reply htdocs/comm/propal/list.php:551]: Yes, if you compare comma-join with left join without considering the sorting arguments.

 But in this case, the statement was:

```
... , llx_societe_commerciaux as sc
...
WHERE...
AND s.rowid = sc.fk_soc
```

Which has *exactly* the same result as, and is more understandeable than : 

```
LEFT JOIN llx_societe_commerciaux as sc ON s.rowid = sc.fk_soc
```

Repeating myself : Please note that **comma-join is NOT equivalent to inner-join"". "The precedence of the comma operator is less than that of INNER JOIN, CROSS JOIN, LEFT JOIN, and so on" [source on MySQL manual](https://dev.mysql.com/doc/refman/8.0/en/join.html)

File: htdocs/commande/list.php (hunk 2 above)
Comment (thomas-Ngr) [reply htdocs/commande/list.php:452]: OK, I guess I will simply replace every comma-join by an explicit INNER JOIN.

**Comment by thomas-Ngr** [comment]
@hregis, I don't see any comma-join in this graph, which is the point here.

**Comment by thomas-Ngr** [comment]
@eldy OK, I reworked this PR using only INNER JOINs.

File: htdocs/commande/list.php (hunk 2 above)
Comment (eldy) [reply htdocs/commande/list.php]: > OK, I guess I will simply replace every comma-join by an explicit INNER JOIN.

Yes,for similar result, comma-join can be replaced with inner join. It is the other way to write inner join.

**Comment by eldy** [comment]
As it is a re-engeneering code, it must be pushed into develop.

**Comment by thomas-Ngr** [comment]
@eldy I close this PR and open this one for following : #26530

**Pull request closed without merging**